data_safe_haven/commands/deploy.py

```diff
@@ -28,6 +31,8 @@ def shm(
     """Deploy a Safe Haven Management component"""
     context = ContextSettings.from_file().assert_context()
     config = Config.from_remote(context)
+    pulumi_config = DSHPulumiConfig.from_remote(context)
```

Comment:
Good point. I'll add something like "get or create DSHPulumiConfig" :+1:.

Reply:
What command creates the remote Pulumi config?

Reply:
To write it to Azure Storage? `DSHPulumiConfig.upload` but you would need to check if one already exists and create an empty instance of the class if not.

Reply:
I'm thinking about the workflow that someone would have when trying to deploy this. Something like:

```
dsh context add # makes the local context
dsh context create # deploys Azure resources
dsh config template > config.yaml # creates a local YAML file
dsh config upload config.yaml # upload the local config to Azure
dsh deploy shm
```

this last line breaks because it looks for a remote Pulumi config but none of the previous commands have created it.

Also - this workflow seems a bit overcomplicated, but we can fix that elsewhere.

Reply:
I prefer to keep the workflow explicit like that. I wouldn't like the command which adds a description of a context to also deploy infrastructure.

Reply:
Fair enough. Perhaps we could replace `dsh config upload config.yaml` with something like `dsh deploy shm --config config.yaml` which would prompt for "are you sure?" if there's a remote config that differs from `config.yaml`.

Reply:
I think that sounds sensible. It would be easy (and probably useful) to keep the `$ dsh config upload ...` command. Both functions should just need to call a couple of `Config` methods.

Could you open an issue, if there isn't one already.

Reply:
If this is in a usable state by this afternoon, we can discuss it during a live run-through.

Reply:
I'm happy to have more powerful advanced commands (or options to commands) but I'd like to keep the minimal required commands simple and easy to understand. If possible, they would just be: `dsh initialise; dsh deploy` but I realise we're not there at the moment!

Reply:
I think this is a philosophical argument.

I would say that isn't simple or easy to understand. Having so few commands means they must be complex and handle a lot of processes without input from the user.

I would say something more like a tree (`dsh shm {deploy, destroy}`, `dsh sre {deploy, destroy}` is simple and easy to understand because the commands are clear, limited and descriptive.

Could you open an issue so I can resolve the conversation here?

Reply:
See #1836 for discussion.